dialog/file.go

```diff
@@ -469,13 +481,16 @@ func (f *fileDialog) setLocation(dir fyne.URI) error {
 			buildDir = "/"
 			d = "/"
 		} else if i > 0 {
-			buildDir = filepath.Join(buildDir, d)
+			buildDir = buildDir + "/" + d
 		} else {
 			d = buildDir
-			buildDir = d + string(os.PathSeparator)
+			buildDir = d + "/"
+		}
+		newURL := dir.Scheme() + "://" + buildDir
```

Comment:
This assumes that a URI  (not URL!) is a scheme followed by a path. This is only true for file URIs. Most contain a host and/or port portion as well.

Reply:
> This assumes that a URI (not URL!) is a scheme followed by a path. This is only true for file URIs. Most contain a host and/or port portion as well.

This variable is actually for a URI, not a URL. I will rename it and check to see if there are other components of the fyne URI with variable name "dir" that need to be included in this constructed URI.

I will create a separate branch for this fix based on your dev branch and resubmit. Give me a little time as I am working on something else right now.

Reply:
I did not write the code you are referring to. I was trying to do the simplest thing possible with rewriting a bunch of existing code. Not sure why the original author did not choose to use storage.Child(). Was it not evailable when this was originally written?

In any case, I can check into that solution as well.

Reply:
> I did not write the code you are referring to. I was trying to do the simplest thing possible without rewriting a bunch of existing code.

I appreciate that, but sometimes when the existing code contains the bug we need to fix that directly instead of following the old pattern and working around it to minimise the change.

> Not sure why the original author did not choose to use storage.Child(). Was it not evailable when this was originally written?

Well, when this only worked with file:// the assumptions held - because it does not (or did not) have any other URI elements like host or port. Those assumptions no longer hold.

Thanks for looking into it.